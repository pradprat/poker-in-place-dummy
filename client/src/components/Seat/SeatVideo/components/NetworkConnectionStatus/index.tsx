import React, { memo, createContext, useContext } from "react";
import { styled } from "@material-ui/core/styles";

export const NetworkStatusVisibilityContext = createContext(false);

const Container = styled("div")({
  display: "flex",
  alignItems: "flex-end",
  "& div": {
    width: "23%",
    border: "1px solid black",
    boxSizing: "content-box",
    "&:not(:last-child)": {
      borderRight: "none",
    },
  },
});

const NetworkConnectionStatus = ({ className }: { className?: string }): JSX.Element => {
  const shouldRender = useContext(NetworkStatusVisibilityContext);

  if (!shouldRender) {
    return null;
  }

  return (
    <Container className={className}>
      {[0, 1, 2, 3].map((level) => (
        <div
          key={level}
          style={{
            height: `${20 * (level + 1)}%`,
            background: "#0c0",
          }}
        />
      ))}
    </Container>
  )
}

export default memo(NetworkConnectionStatus);
