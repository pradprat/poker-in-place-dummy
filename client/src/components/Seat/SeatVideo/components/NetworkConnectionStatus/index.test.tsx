import React from "react";
import renderer from "react-test-renderer";

import NetworkConnectionStatus from ".";

describe("NetworkQualityLevel component", () => {
  it("should render correctly if isConnected", () => {
    const tree = renderer.create(<NetworkConnectionStatus />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
