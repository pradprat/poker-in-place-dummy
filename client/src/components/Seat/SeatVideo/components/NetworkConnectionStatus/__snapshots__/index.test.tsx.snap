// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkQualityLevel component should render correctly if isConnected 1`] = `null`;
